### PR TITLE
Sf expansion 481

### PR DESCRIPTION
**Pull request by AnneONS**
## Pull Request submission

*Insert detailed bullet points about your changes here!*

Instructions:
- Run the pipeline on the sf_expansion branch
- View the output in the R:\BERD Results System Development 2023\DAP_emulation\imputation\imputation_qa folder
- filter the data on short forms (formtype==6)
- Filter to just one imputation class (column "imp_class"). We suggest class "C_AH", but you could try others
- You should see "imputation_***" columns, where *** is the breakdown variable (such as "204" or "303"). **Check the variables in these columns exist**
- For any record you can **check the results** using Excel formulas. 

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #481

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/imputation/expansion_imputation.py [hunk 1]
```diff
@@ -58,6 +58,9 @@ def evaluate_imputed_ixx(
 def split_df_on_trim(df: pd.DataFrame, trim_bool_col: str) -> pd.DataFrame:
     """Splits the dataframe in based on if it was trimmed or not"""
 
+    # TODO: remove this temporary fix to cast Nans to False
+    df[trim_bool_col].fillna(False, inplace=True)
```
Comment (dpurches) [inline src/imputation/expansion_imputation.py:62]: @AnneONS does this need to be/can this be removed before we merge?

Comment (AnneONS) [reply src/imputation/expansion_imputation.py:62]: no, I'm not sure how we would immediately, so shall we leave it in?

Comment (dpurches) [reply src/imputation/expansion_imputation.py:62]: Yeh that's fine - so long as it doesn't have any impact downstream

Comment (AnneONS) [reply src/imputation/expansion_imputation.py:62]: no, it won't

**Review by dpurches — APPROVED** [review]
Looks great! And the numbers all add up in the outputs (at last!). 

Only question from me @AnneONS is do we need to update the `imputation_marker` column for short form expansion?

**Comment by AnneONS** [comment]
> Looks great! And the numbers all add up in the outputs (at last!).
> 
> Only question from me @AnneONS is do we need to update the `imputation_marker` column for short form expansion?

I'd say no as ALL short forms undergo sf_expansion, so there's no point. I suggest imp_marker is just where we do TMI, MoR actual imputation (sf expansion is called imputation, but it's rather different I'd say) I'll ask Ana

**Comment by AnneONS** [comment]
> > Looks great! And the numbers all add up in the outputs (at last!).
> > Only question from me @AnneONS is do we need to update the `imputation_marker` column for short form expansion?
> 
> I'd say no as ALL short forms undergo sf_expansion, so there's no point. I suggest imp_marker is just where we do TMI, MoR actual imputation (sf expansion is called imputation, but it's rather different I'd say) I'll ask Ana

Confirmed with Ana, she agrees.

**Comment by dpurches** [comment]
> > > Looks great! And the numbers all add up in the outputs (at last!).
> > > Only question from me @AnneONS is do we need to update the `imputation_marker` column for short form expansion?
> > 
> > 
> > I'd say no as ALL short forms undergo sf_expansion, so there's no point. I suggest imp_marker is just where we do TMI, MoR actual imputation (sf expansion is called imputation, but it's rather different I'd say) I'll ask Ana
> 
> Confirmed with Ana, she agrees.

Makes sense to me!

**Pull request merged**